dbms/src/AggregateFunctions/AggregateFunctionTopK.h

```diff
@@ -119,7 +127,14 @@ class AggregateFunctionTopK
 /// Generic implementation, it uses serialized representation as object descriptor.
 struct AggregateFunctionTopKGenericData
 {
-    using Set = SpaceSaving<StringRef, StringRefHash>;
+    using Set = SpaceSaving
+    <
+        std::string,
+        StringRef,
+        StringRefHash,
+        HashTableGrower<TOP_K_DEFAULT_DEGREE>,
+        HashTableAllocatorWithStackMemory<sizeof(StringRef) * (1 << (TOP_K_DEFAULT_DEGREE/2))>
```

Comment:
It is reasonable to have stack space equal to initial size of hash table in bytes.
(So, it will use stack memory before first resize.)

You are using `HashTableGrower<TOP_K_DEFAULT_DEGREE>`.
It means, that initial size of hash table is `1 << TOP_K_DEFAULT_DEGREE`.
With TOP_K_DEFAULT_DEGREE = 10 it will be 1024 elements (seems quite large, I recommend something about 8 or 16 elements).
It will take `1024 * sizeof(Cell)` bytes.

And you need to pass this value as template parameter for `HashTableAllocatorWithStackMemory`.

Example is here:
https://github.com/yandex/ClickHouse/blob/master/dbms/src/AggregateFunctions/AggregateFunctionUniq.h#L89